cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -439,6 +440,99 @@ public Broker broker(int brokerId) {
     return rack == null ? null : rack.broker(brokerId);
   }
 
+  /**
+   * Ask the cluster model to keep track of the replicas sorted with the given score function.
+   *
+   * It is recommended to use the functions from {@link ReplicaSortFunctionFactory} so the functions can be maintained
+   * in a single place.
+   *
+   * The sorted replica will only be updated in the following cases:
+   * 1. A replica is added to a broker
+   * 2. A replica is removed from a broker
+   * 3. A replica's role has changed from leader to follower, and vice versa.
+   *
+   * The sorted replicas are named using the given sortName, and can be accessed using
+   * {@link Broker#trackedSortedReplicas(String)}. If the sorted replicas are no longer needed,
+   * {@link #untrackSortedReplicas(String)} to release memory.
+   *
+   * @param sortName the name of the sorted replicas.
+   * @param scoreFunction the score function to sort the replicas with the same priority.
+   * @see SortedReplicas
+   */
+  public void trackSortedReplicas(String sortName, Function<Replica, Double> scoreFunction) {
```

Comment:
It is probably better to have separate java docs. It is not necessary to point people to another method that has more parameters. Also, when the java doc is generated, each method should have their own java doc. Otherwise, users may have to click around to find the java doc.